src/map/itemdb.cpp

```diff
@@ -1536,133 +1534,205 @@ char itemdb_isidentified(t_itemid nameid) {
+
+			for (const YAML::Node &listit : listNode) {
+				std::string item_name;
+
+				if (!this->asString(listit, "Item", item_name))
+					continue;
+
+				struct item_data *item = itemdb_search_aegisname(item_name.c_str());
+
+				if (item == nullptr) {
+					this->invalidWarning(listit["Item"], "Unknown Item %s.\n", item_name.c_str());
+					continue;
+				}
+
+				auto entry = std::make_shared<s_item_group_entry>();
+				entry->nameid = item->nameid;
+
+				if (this->nodeExists(listit, "Rate")) {
+					uint16 rate;
+
+					if (!this->asUInt16(listit, "Rate", rate))
+						continue;
+
+					entry->rate = rate;
+				} else {
+					entry->rate = 0;
+				}
+
+				if (subgroup == 0 && entry->rate > 0) {
+					this->invalidWarning(listit["Item"], "SubGroup 0 is reserved for item without rate ('must' item). Defaulting to 0.\n");
+					entry->rate = 0;
+				}
+				if (subgroup != 0 && entry->rate == 0) {
+					this->invalidWarning(listit["Item"], "Entry must have a probability for group above 0, skipping.\n");
+					continue;
+				}
+
+				if (this->nodeExists(listit, "Amount")) {
+					uint16 amount;
+
+					if (!this->asUInt16(listit, "Amount", amount))
+						continue;
+
+					entry->amount = cap_value(amount, 1, MAX_AMOUNT);
+				} else {
+					entry->amount = 1;
+				}
+
+				if (this->nodeExists(listit, "Duration")) {
+					uint16 duration;
+
+					if (!this->asUInt16(listit, "Duration", duration))
+						continue;
+
+					entry->duration = duration;
+				} else {
+					entry->duration = 0;
+				}
+
+				if (this->nodeExists(listit, "Announced")) {
+					bool isAnnounced;
+
+					if (!this->asBool(listit, "Announced", isAnnounced))
+						continue;
+
+					entry->isAnnounced = isAnnounced;
+				} else {
+					entry->isAnnounced = false;
+				}
+
+				if (this->nodeExists(listit, "GUID")) {
+					bool guid;
+
+					if (!this->asBool(listit, "GUID", guid))
+						continue;
+
+					entry->GUID = guid;
+				} else {
+					entry->GUID = false;
+				}
+
+				if (this->nodeExists(listit, "Named")) {
+					bool named;
+
+					if (!this->asBool(listit, "Named", named))
+						continue;
+
+					entry->isNamed = named;
+				} else {
+					entry->isNamed = false;
+				}
+
+				if (this->nodeExists(listit, "Bound")) {
+					std::string bound_name;
+
+					if (!this->asString(listit, "Bound", bound_name))
+						continue;
+
+					std::string bound_name_constant = "BOUND_" + bound_name;
+					int64 bound;
+
+					if (!script_get_constant(bound_name_constant.c_str(), &bound) || bound < BOUND_NONE || bound >= BOUND_MAX) {
+						this->invalidWarning(listit["Group"], "Invalid Bound %s.\n", bound_name.c_str());
+						continue;
+					}
+
+					entry->bound = static_cast<uint8>(bound);
+				} else {
+					entry->bound = BOUND_NONE;
+				}
+
+				std::shared_ptr<s_item_group_random> random = util::umap_find(group->random, subgroup);
+
+				if (random == nullptr) {
+					random = std::make_shared<s_item_group_random>();
+					group->random[subgroup] = random;
+				}
```

Comment:
All this should be moved above otherwise the import will not work correctly.
Additionally it is searched on each iteration instead of once only.